### ISSUE TITLE
Normalisation of depth data and depth prediction performance

### ISSUE THREAD

**Issue by fbragman** [lorenmt/auto-lambda#6]
Hi,

Congratulations on a great paper :) 

Thanks a lot for making your code open-source. I went through your code and it seems like you normalise the input RGB data to `[-1, 1]` scale whilst the depth data is normalised to `[-1, max]`. It was my understanding that for Cityscapes, the depth data would be normalised to `[-1, 1]` after using the `map_disparity` function and the RGB data normalised to ImageNet stats if for instance using pre-trained weights. Am I wrong?

I also have a general question about training depth prediction models from Cityscapes. I have tried various flavours of models (DeepLabV3, HRNet) and yet, training single-task depth prediction network seems to yield overly smooth depth maps with convergence of the loss occurring very soon in training regardless of the learning rate (1e-3, 1e-4 etc. for ADAM). For reference, the RGB data is normalised using ImageNet stats (using pre-trained encoders on ImageNet) and the depth data is either normalised to `[-1, 1]` or `[-1, max]` (using your disparity mapping functions) 

I was wondering if you could comment based on your experience on the dataset? This would be very helpful. These same networks have been tested on the 19-class segmentation problem.

Many thanks

**Comment by lorenmt** [comment]
Hello,

The depth data are not normalised. Here the -1 represents invalid depths, which will be masked during backprop. The depth measurements are either in true values measured in meters (in NYUv2), or inverse meters (in Cityscapes).

For fairness, all methods here were trained from scratch and thus we did not apply any ImageNet normalisation, and thus the images were normalised to -1 to 1.

For your general question, yes for single task depth estimation, the performance would be pretty poor because the dataset has a consistent distribution for street-view depth: the middle part is farther and the edge is closer. Therefore, it's very easy to overfit on training data and leads to over-smooth prediction.  That's how the multi-task learning method is very useful in this case.

**Comment by fbragman** [comment]
Thanks a lot for the response! That's really helpful.

**Comment by prismformore** [comment]
@lorenmt Hi Shikun, thank you for your interesting work. May I know whether you paired the depth with semantic labels as MTAN did? It looks like you didn't do that in the preprocessing code.

**Comment by lorenmt** [comment]
Hi, Sorry I don't understand your question. Could you specify what did you mean by "data paring"? We used the exact same training data in NYUv2 from MTAN codebase to train Auto-Lambda.

**Comment by prismformore** [comment]
> Hi, Sorry I don't understand your question. Could you specify what did you mean by "data paring"? We used the exact same training data in NYUv2 from MTAN codebase to train Auto-Lambda.


@lorenmt  Thank you for your prompt reply. I mean the depth and semantics paring in:
"We pair the depth estimation task with three levels of semantic segmentation using 2, 7 or 19 classes (excluding the void group in 7 and 19 classes). Labels for the 19 classes and the coarser 7 categories are defined as in the original CityScapes dataset." quoted from MTAN paper. 

I thought you only evaluate depth with the interested semantic objects. Maybe I misunderstand the words here.

So if you use exactly the same depth definition in both MTAN and auto-lambda on the Cityscape dataset, may I know why the disparity accuracy in MTAN (0.01~) is better than that in auto-lambda (0.80~)? Is it because of the difference in resolution?

Thank you for your help!

**Comment by lorenmt** [comment]
@prismformore I see your point here. I believe you should look into this function: https://github.com/lorenmt/auto-lambda/blob/e1bed93b753002fdbcfb0ee213def2c2824d866a/create_dataset.py#L215, which should be the reason of the discrepancy of depth error between MTAN and auto-lambda. This function gives the more mathematically accurate implementation of true disparity value in the Cityscapes dataset.., as an issue reported here: https://github.com/lorenmt/mtan/issues/25.

And I removed coarser semantic classes, just keeping the standard 19-class in Cityscapes.

All data in NYUv2 stays the same.

Hope that resolves your question.

**Comment by prismformore** [comment]
@lorenmt Thank you for your insights! This is super helpful.
